src/Microsoft.TemplateEngine.Edge/Settings/GlobalSettings.cs

```diff
@@ -0,0 +1,150 @@
+
+        public async Task<IDisposable> LockAsync(CancellationToken token)
+        {
+            if (_disposed)
+            {
+                throw new ObjectDisposedException(nameof(GlobalSettings));
+            }
+            token.ThrowIfCancellationRequested();
+            // We must use Mutex because we want to lock across different processes that might want to modify this settings file
+            var mutex = await AsyncMutex.WaitAsync($"812CA7F3-7CD8-44B4-B3F0-0159355C0BD5{_globalSettingsFile}".Replace("\\", "_").Replace("/", "_"), token, Unlocked).ConfigureAwait(false);
+            _locked = true;
+            return mutex;
+        }
+
+        private void Unlocked()
+        {
+            _locked = false;
+        }
+
+        public void Dispose()
+        {
+            if (_disposed)
+            {
+                return;
+            }
+            _disposed = true;
+
+            _watcher?.Dispose();
+            _watcher = null;
+        }
+
+        public async Task<IReadOnlyList<TemplatePackageData>> GetInstalledTemplatesPackagesAsync(CancellationToken cancellationToken)
+        {
+            if (_disposed)
+            {
+                throw new ObjectDisposedException(nameof(GlobalSettings));
+            }
+
+            if (!_environmentSettings.Host.FileSystem.FileExists(_globalSettingsFile))
+            {
+                return Array.Empty<TemplatePackageData>();
+            }
+
+            for (int i = 0; i < 5; i++)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+
+                try
+                {
+                    string? textFileContent = _paths.ReadAllText(_globalSettingsFile, "{}");
+                    GlobalSettingsData? data = JsonConvert.DeserializeObject<GlobalSettingsData>(textFileContent);
+                    return data.Packages ?? Array.Empty<TemplatePackageData>();
+                }
+                catch (Exception)
+                {
+                    if (i == 4)
+                    {
+                        throw;
+                    }
+                }
+                await Task.Delay(20).ConfigureAwait(false);
+            }
+            throw new InvalidOperationException();
+        }
+
+        public async Task SetInstalledTemplatesPackagesAsync(IReadOnlyList<TemplatePackageData> packages, CancellationToken cancellationToken)
+        {
+            if (_disposed)
+            {
+                throw new ObjectDisposedException(nameof(GlobalSettings));
+            }
+
+            if (!_locked)
+            {
+                throw new InvalidOperationException($"Before calling {nameof(SetInstalledTemplatesPackagesAsync)}, {nameof(LockAsync)} must be called.");
+            }
+
+            string? serializedText = JsonConvert.SerializeObject(new GlobalSettingsData()
+            {
+                Packages = packages
+            });
+
+            for (int i = 0; i < 5; i++)
```

Comment:
Why 5 attempts? Make it a constant declared as a field on the class. Use `while (true)` so you don't have to worry what happens after the loop and the number of attempts is only in place (now 5 here and 4 in the `if` below).

Reply:
Added constant.